supporting_scripts/generate_code_cov_table/generate_code_cov_table.py

```diff
@@ -31,29 +34,89 @@ def environ_or_required(key, required=True):
     )
 
 
-def get_client_tests_cov_report_url(build_id):
-    return f"{base_url}/browse/{project_key}-{plan_key}{build_id}/latest/artifact/shared/Coverage-Report-Client-Tests"
+def download_and_extract_zip(url, headers):
+    if url is None:
+        return None
+    try:
+        response = requests.get(url, headers=headers)
```

Comment:
Would be nice if you could also use tqdm here, as this is one of the longest running operation of this script